### DISCUSSION TITLE
query error

### DISCUSSION THREAD

**Discussion by michelkluger** [Tsadoq/kompy#13] [Q&A]
2024-01-17 23:43,857 Response: {"status":400,"error":"Validation","message":"query is required for type=tour_planned"}

what could be the cause?
my script

```
import gpxpy

count_errors = 0
files_already_attempted = 0
files_successfully_uploaded = 0

for file in files:
    filename = f"{current_dir}\{file}"
    
    with open(filename, 'r') as gpx_file:
        try:
            gpx = gpxpy.parse(gpx_file)
        except:
            print('error')
            count_errors += 1
        success_bool = connector.upload_tour(
            tour_object=gpx,
            activity_type='mtb_easy',
            tour_name=file,
            time_in_motion = 42,
        )
        files_already_attempted += 1
        if success_bool:
            files_successfully_uploaded += 1
        print(success_bool)
```

**Comment by michelkluger** [comment]
update, the tours are being upload despite the error 400, but on completed tours, I would like to have them as planned tours, could that be a flag in the upload?

**Reply by Tsadoq** [reply]
Not that I am aware of. Have you tried not setting the time in motion? 

https://static.komoot.de/doc/external-api/v007/index.html#tours__data_type_fit_post

The Komoot documentation doesn't specify that param but I will check. The field is present in their tour object by Komoot (if you download one of your tours you'll see it) but not in the request for the upload.

Given that GPX file are at the end of the day XML files I'd try to add a tag for those two fields and see how it goes. If you do not have the time I'm gonna do that in the weekend.

Btw,does the GPX file come from Komoot? Would you feel ok at sharing it?

**Reply by michelkluger** [reply]
I was trying to bulk upload those tours, 

https://cycletour.org.nz/assets/gps/gpx/tracks.gpx.zip

**Reply by michelkluger** [reply]
so I have them in komoot without having to manually do it for 100+ files

**Reply by michelkluger** [reply]
thank you very much for digging into that, and for creating a very cool open source project 

in the api endpoint for upload there is indeed no mentioned for any params of tour_type planned or completed,

https://static.komoot.de/doc/external-api/v007/index.html#tours__data_type_fit_post

**Reply by michelkluger** [reply]
I also tried adding the time in motion in the upload function, with similar results
